src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -268,3 +299,37 @@ function validate_connectivity(M, nodes, bus_lookup)
     end
     return connected
 end
+
+"""
+Computes the connected components of the network graph using network traversal.
+Returns a set of sets of buses, each set is a connected component
+"""
+# this function borrows heavily from "calc_connected_compoents" in PowerModels.jl
+function find_connected_components(sys::System)
+    buses = get_components(Bus, sys, x -> get_bustype(x) != BusTypes.ISOLATED)
+    branches = get_components(Branch, sys, get_available)
+
+    return find_connected_components(branches, buses)
+end
+
+function find_connected_components(branches, buses)
```

Comment:
This code is already implemented in the PowerFlow calculations. We need to find similarities and avoid duplicates.